src/Umbraco.Web.UI.Client/src/views/languages/edit.controller.js

```diff
@@ -20,150 +19,118 @@
-                vm.labels.noFallbackLanguageOption = values[6];
-                vm.labels.areYouSure = values[9];
-
-                $scope.properties = {
-                    fallbackLanguage: {
-                        alias: "fallbackLanguage",
-                        description: values[7],
-                        label: values[8]
-                    }
-                };
+                vm.labels.addLanguage = values[1];
+                vm.labels.areYouSure = values[2];
 
                 if ($routeParams.create) {
-                    vm.page.name = vm.labels.addLanguage;
+                    vm.isNew = true;
+                    vm.language.name = vm.labels.addLanguage;
                 }
-            });
-
-            vm.loading = true;
-
-            var promises = [];
+            }));
 
-            //load all culture/languages
+            // Load all culture/languages
             promises.push(languageResource.getCultures().then(function (culturesDictionary) {
-                var cultures = [];
-                angular.forEach(culturesDictionary, function (value, key) {
-                    cultures.push({
-                        name: key,
-                        displayName: value
-                    });
-                });
-                vm.availableCultures = cultures;
+                vm.availableCultures = culturesDictionary;
             }));
 
-            //load all possible fallback languages
+            // Load all possible fallback languages
             promises.push(languageResource.getAll().then(function (languages) {
                 vm.availableLanguages = languages.filter(function (l) {
                     return $routeParams.id != l.id;
                 });
-                vm.loading = false;
             }));
 
             if (!$routeParams.create) {
-
                 promises.push(languageResource.getById($routeParams.id).then(function (lang) {
                     vm.language = lang;
 
-                    vm.page.name = vm.language.name;
-
-                    /* we need to store the initial default state so we can disable the toggle if it is the default.
-                    we need to prevent from not having a default language. */
+                    // We need to store the initial default state so we can disable the toggle if it is the default.
+                    // We need to prevent from not having a default language.
                     vm.initIsDefault = Utilities.copy(vm.language.isDefault);
 
                     makeBreadcrumbs();
 
-                    //store to check if we are changing the lang culture
-                    currCulture = vm.language.culture;
+                    // Store to check if we are changing the lang culture
+                    initCulture = vm.language.culture;
                 }));
             }
 
-            $q.all(promises, function () {
+            $q.all(promises).then(function () {
                 vm.loading = false;
             });
```

Comment:
The callback function was never executed, that's probably why the loading indicator was disabled after all languages were loaded.